runtime/ops/http.rs

```diff
@@ -409,6 +410,9 @@ async fn op_http_response(
   })
   .await?;
 
+  if maybe_response_body_rid.is_none() {
+    conn_resource.deno_service.waker.wake();
+  }
```

Comment:
I mean, sure, whatever makes the tests pass.

But I wonder if the waker.wake() could be done more centrally inside `conn_resource.poll` but only when it's about to return pending...

Reply:
> But I wonder if the waker.wake() could be done more centrally inside conn_resource.poll but only when it's about to return pending...

But we don't want that inside `conn_resource.poll` - if we did that in `conn_resource.poll` then we'd be waking the future of `op_http_request_next` while polling that op. Actually we want to wake `op_http_request_next` when a response has finished writing which can happen either in `op_http_response` or `op_http_response_close `.